src/utils/createNoStringParameterToFunctionWalker.ts

```diff
@@ -0,0 +1,210 @@
+
+                if (node.body !== undefined) {
+                    scope.addGlobalScope(node.body, node.getSourceFile(), options);
+                }
+
+                ts.forEachChild(node, cb);
+                scope = scope.parent;
+            } else if (tsutils.isClassDeclaration(node)) {
+                const classScope = (scope = new Scope(scope));
+                node.members.forEach(
+                    (element: ts.ClassElement): void => {
+                        const prefix: string = AstUtils.isStatic(element) && node.name !== undefined ? node.name.getText() + '.' : 'this.';
+
+                        if (element.kind === ts.SyntaxKind.MethodDeclaration) {
+                            // add all declared methods as valid functions
+                            classScope.addFunctionSymbol(prefix + (<ts.MethodDeclaration>element).name.getText());
+                        } else if (element.kind === ts.SyntaxKind.PropertyDeclaration) {
+                            const prop: ts.PropertyDeclaration = <ts.PropertyDeclaration>element;
+                            // add all declared function properties as valid functions
+                            if (AstUtils.isDeclarationFunctionType(prop)) {
+                                classScope.addFunctionSymbol(prefix + (<ts.MethodDeclaration>element).name.getText());
+                            } else {
+                                classScope.addNonFunctionSymbol(prefix + (<ts.MethodDeclaration>element).name.getText());
+                            }
+                        }
+                    }
+                );
+                ts.forEachChild(node, cb);
+                scope = scope.parent;
+            } else if (tsutils.isFunctionDeclaration(node)) {
+                scope = new Scope(scope);
+                scope.addParameters(node.parameters);
+                ts.forEachChild(node, cb);
+                scope = scope.parent;
+            } else if (tsutils.isConstructorDeclaration(node)) {
+                scope = new Scope(scope);
+                scope.addParameters(node.parameters);
+                ts.forEachChild(node, cb);
+                scope = scope.parent;
+            } else if (tsutils.isMethodDeclaration(node)) {
+                scope = new Scope(scope);
+                scope.addParameters(node.parameters);
+                ts.forEachChild(node, cb);
+                scope = scope.parent;
+            } else if (tsutils.isArrowFunction(node)) {
+                scope = new Scope(scope);
+                scope.addParameters(node.parameters);
+                ts.forEachChild(node, cb);
+                scope = scope.parent;
+            } else if (tsutils.isFunctionExpression(node)) {
+                scope = new Scope(scope);
+                scope.addParameters(node.parameters);
+                ts.forEachChild(node, cb);
+                scope = scope.parent;
+            } else if (tsutils.isSetAccessorDeclaration(node)) {
```

Comment:
A bunch of these conditions are the same. Please switch to a single `if` statement with `||`s for the different cases to reduce code duplication.